Source/Diagnostics/ReducedDiags/ParticleHistogram.cpp

```diff
@@ -197,7 +199,9 @@ void ParticleHistogram::ComputeDiags (int step)
                     auto const uz = d_uz[i] / PhysConst::c;
 
                     // don't count a particle if it is filtered out
-                    if (!fun_filterparser(t, x, y, z, ux, uy, uz)) return;
+                    if (do_parser_filter)
+                        if (!fun_filterparser(t, x, y, z, ux, uy, uz))
```

Comment:
@WeiqunZhang I am not sure if we can make this safer (for developers): If the user did not specify a filter, the nullptr passed to `getParser()` seems to return an object that segfaults when evaluated.